### PR TITLE
Getting ready for production

### PR DESCRIPTION
**Pull request by enricostano**
### WAT

In production Nginx will proxy TCP port `80` to `127.0.0.1:8080` (Express) and the Rails API will be in UNIX socket.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
       "env": {
         "NODE_PATH": "./src",
         "NODE_ENV": "production",
-        "PORT": 8080,
+        "SOCK_PATH": "/var/run/node/express.sock",
```
Comment (andresgutgon) [inline package.json]: I don't get it. It's the api who has to run by sockets. Or both?

Comment (enricostano) [inline package.json]: Both, will takle API socket in production in other PR

Comment (andresgutgon) [inline package.json]: got it

**Comment by andresgutgon** [comment]
:+1:

**Comment by enricostano** [comment]
@andresgutgon can you please confirm that the `ApiClient` library is used only in the browser to make XHR requests and we don't use it directly from Express.js?

EDIT: Related https://www.quora.com/Does-Node-js-utilize-XMLHttpRequest

**Comment by sauloperez** [comment]
:+1:

File: src/helpers/api/apiHttp.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    ['get', 'post', 'put', 'patch', 'del'].forEach((method) => {
+      this[method] = (path) => new Promise((resolve, reject) => {
+        const request = request[method](formatUrl(path));
+
+        if (params) {
+          request.query(params);
+        }
+
+        if (user_id) {
+          request.set('X-katuma-user-id', user_id);
```
Comment (andresgutgon) [inline src/helpers/api/apiHttp.js]: `user_id || ''`

File: src/helpers/api/apiHttp.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import request from 'request';
+import config from '../../config';
+
+function formatUrl(path) {
+  const adjustedPath = path[0] !== '/' ? '/' + path : path;
+  const apiPath = `/api/${config.apiVersion}${adjustedPath}`;
+  const apiDomain = `http://${(process.env.HOST || 'localhost')}:${config.apiPort}`;
```
Comment (andresgutgon) [inline src/helpers/api/apiHttp.js]: WIP sockect

File: src/server.js [hunk 4]
```diff
@@ -62,6 +62,8 @@ const proxy = httpProxy(`http://${(process.env.HOST || 'localhost')}`, {
     if (req.session.user_id) {
       req.headers['X-katuma-user-id'] = req.session.user_id;
     }
+    req.socketPath = config.socketPath;
```
Comment (andresgutgon) [inline src/server.js]: Maybe configure this not by request

File: src/config.js [hunk 5]
```diff
@@ -13,9 +13,23 @@ const environment = {
...
+    options.socketPath = process.env.API_SOCK_PATH;
+  }
+
+  return options;
+}
+
 module.exports = Object.assign({
-  port: process.env.PORT,
-  apiPort: process.env.APIPORT,
+  listen_to: process.env.SOCK_PATH || process.env.PORT,
```
Comment (andresgutgon) [inline src/config.js]: move this to `server.js`

File: src/helpers/api/apiAjax.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  const adjustedPath = path[0] !== '/' ? '/' + path : path;
+  const apiPath = `/api/${config.apiVersion}${adjustedPath}`;
+
+  return apiPath;
+}
+
+class ApiAjax {
+  constructor() {
+    ['get', 'post', 'put', 'patch', 'del'].forEach((method) =>
+      this[method] = (path, { params, data } = {}) => new Promise((resolve, reject) => {
```
Comment (enricostano) [inline src/helpers/api/apiAjax.js]: What about `params`?

**Comment by enricostano** [comment]
@andresgutgon can you please help me rebase this? So we can move forward and merge it.

**Comment by andresgutgon** [comment]
:muscle:

**Pull request merged**